transports/webrtc/tests/smoke.rs

```diff
@@ -95,19 +95,28 @@ async fn smoke() -> Result<()> {
         e => panic!("{:?}", e),
     }
 
-    match a.next().await {
-        Some(SwarmEvent::IncomingConnection { .. }) => {}
-        e => panic!("{:?}", e),
-    };
+    let pair = select(a.next(), b.next());
+    match pair.await {
+        Either::Left((Some(SwarmEvent::IncomingConnection { .. }), _)) => {}
+        Either::Left((e, _)) => panic!("{:?}", e),
+        Either::Right(_) => panic!("b completed first"),
+    }
 
-    let pair = join(a.next(), b.next());
+    let pair = select(a.next(), b.next());
     match pair.await {
-        (
-            Some(SwarmEvent::ConnectionEstablished { .. }),
-            Some(SwarmEvent::ConnectionEstablished { .. }),
-        ) => {}
-        e => panic!("{:?}", e),
-    };
+        Either::Left((Some(SwarmEvent::ConnectionEstablished { .. }), _)) => {}
+        Either::Left((e, _)) => panic!("{:?}", e),
+        Either::Right((Some(SwarmEvent::ConnectionEstablished { .. }), _)) => {}
+        Either::Right((e, _)) => panic!("{:?}", e),
+    }
+
+    let pair = select(a.next(), b.next());
+    match pair.await {
+        Either::Left((Some(SwarmEvent::ConnectionEstablished { .. }), _)) => {}
+        Either::Left((e, _)) => panic!("{:?}", e),
+        Either::Right((Some(SwarmEvent::ConnectionEstablished { .. }), _)) => {}
+        Either::Right((e, _)) => panic!("{:?}", e),
+    }
```

Comment:
It hurts me to see that writing tests against one or multiple swarms is this difficult and error prone. I've been meaning to write some common utilities for testing swarms and behaviours for a while. I'll put this higher up in my priority list! :)

Reply:
The way tests are written right now is also not intuitive (i.e. it's fine to poll single swarm for Dialing event, but for ConnectionEstablished one need to poll both swarms). Common utilities sound great 👍